test/mjml-props.test.tsx

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+
+import * as mjmlComponents from "../src";
+import { renderToMjml } from "../src/utils/renderToMjml";
+
+/**
+ * Test focused on prop values. The component used for testing the prop
+ * is arbitrary.
+ */
+describe("mjml components prop values", () => {
+  it("should render component with dangerouslySetInnerHTML prop", () => {
+    expect(
+      renderToMjml(
+        <mjmlComponents.MjmlRaw
+          // eslint-disable-next-line react/no-danger
+          dangerouslySetInnerHTML={{ __html: "<div>hello World</div>" }}
+        />
+      )
+    ).toBe(`<mj-raw><div>hello World</div></mj-raw>`);
+  });
+
+  it("incorrect prop value should throw TS error", () => {
+    expect(
+      renderToMjml(
+        <mjmlComponents.MjmlRaw
+          // @ts-expect-error prop should be invalid for test purposes
+          iAmNotAProp="random"
+        />
+      )
+    ).toBe(`<mj-raw i-am-not-a-prop="random"></mj-raw>`);
+  });
+
+  it("fullWidth and inline props convert to string", () => {
+    expect(renderToMjml(<mjmlComponents.MjmlSection fullWidth />)).toBe(
+      `<mj-section full-width="full-width"></mj-section>`
+    );
+    expect(renderToMjml(<mjmlComponents.MjmlStyle inline />)).toBe(
+      `<mj-style inline="inline"></mj-style>`
+    );
+  });
+
+  it("enum prop type accepts all enum values", () => {
+    const { MjmlTable } = mjmlComponents;
+    expect(renderToMjml(<MjmlTable align="left" />)).toBe(
+      `<mj-table align="left"></mj-table>`
+    );
+    expect(renderToMjml(<MjmlTable align="right" />)).toBe(
+      `<mj-table align="right"></mj-table>`
+    );
+    expect(renderToMjml(<MjmlTable align="center" />)).toBe(
+      `<mj-table align="center"></mj-table>`
+    );
+
+    // Test with invalid props. Code should still render but typescript should
+    // throw an error
+    // @ts-expect-error invalid align prop for test purposes
+    expect(renderToMjml(<MjmlTable align="unknown" />)).toBe(
+      `<mj-table align="unknown"></mj-table>`
+    );
+  });
+
+  it("padding accepts px, percent, and ephemeral units", () => {
+    const { MjmlButton } = mjmlComponents;
+    const buttonNoUnit = <MjmlButton padding="0">No unit</MjmlButton>;
+    const buttonPx = <MjmlButton letterSpacing="1px">Px</MjmlButton>;
+    const buttonPercentage = <MjmlButton height="2%">Percent</MjmlButton>;
+    const buttonEm = <MjmlButton letterSpacing="3em">Em</MjmlButton>;
+
+    expect(renderToMjml(buttonNoUnit)).toBe(
+      '<mj-button padding="0">No unit</mj-button>'
+    );
+    expect(renderToMjml(buttonPx)).toBe(
+      '<mj-button letter-spacing="1px">Px</mj-button>'
+    );
+    expect(renderToMjml(buttonPercentage)).toBe(
+      '<mj-button height="2%">Percent</mj-button>'
+    );
+    expect(renderToMjml(buttonEm)).toBe(
+      '<mj-button letter-spacing="3em">Em</mj-button>'
+    );
```

Comment:
Is there a type failure case for the `px, percent, and ephemeral units`?
What type safety does it give us?

Reply:
We have type safety via the PR @hchiufaire added https://github.com/Faire/mjml-react/pull/38

So a typo like "3en" should cause a type error.